s3/src/main/scala/akka/stream/alpakka/s3/impl/SplitAfterSizeWithContext.scala

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright (C) 2016-2020 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.stream.alpakka.s3.impl
+
+import akka.annotation.InternalApi
+import akka.stream.scaladsl.{Flow, SubFlow}
+import akka.stream.stage.{GraphStage, GraphStageLogic, InHandler, OutHandler}
+import akka.stream.{Attributes, FlowShape, Inlet, Outlet}
+import akka.util.ByteString
+import com.github.ghik.silencer.silent
+
+/**
+ * Internal Api
+ *
+ * Splits up a byte stream source into sub-flows of a minimum size while maintaining a Context. Does not attempt to
+ * create chunks of an exact size. Unlike `SplitAfterSize` there is no maximum size since this would imply splitting
+ * up a Context which is not possible to do if the Context is generic.
+ *
+ * This also means that `SplitAfterSizeContext` currently doesn't support buffering since that would require a way to
+ * serialize the Context in the case of a disk buffer which is currently unsupported (we would have to add in a
+ * C => ByteString serializer in the public API)
```

Comment:
That could potentially be deferred to the Akka serialization infrastructure though, and not require any additional parameters in the API, but probably best if it was opt in anyway in that case, and ok to keep in mem like this for now and add once there is a need rather than to make this PR even more complex.

Reply:
Wasn't aware there is an akka serialization infrustructure (I was just thinking of providing a `Flow[C, ByteString, NotUsed]` but yes we are in agreement this can be added later.